### PR TITLE
Remove depreciated #39

### PR DESCRIPTION
**Pull request by annawoodard**

### CODE REVIEW

File: libsubmit/providers/local/local.py [hunk 1]
```diff
@@ -92,7 +92,13 @@ def __init__ (self, config, channel_script_dir=None, channel=None):
              - Config (dict): Dictionary with all the config options.
         '''
 
-        self.channel = channel
+        if channel is None:
+            if channel_script_dir is None:
+                self.channel = LocalChannel(scriptDir=channel_script_dir)
```
Comment (yadudoc) [inline libsubmit/providers/local/local.py]: Maybe reverse the if statement ?

File: libsubmit/providers/local/local.py [hunk 2]
```diff
@@ -92,7 +92,13 @@ def __init__ (self, config, channel_script_dir=None, channel=None):
...
+            else:
+                self.channel = LocalChannel()
+        else:
+            self.channel = channel
```
Comment (yadudoc) [inline libsubmit/providers/local/local.py]: Maybe put a note about why we create the channel in the provider where the general pattern is to have the 
provider be passed a channel from the execution provider factory.

File: libsubmit/providers/local/local.py (hunk 1 above)
Comment (annawoodard) [reply libsubmit/providers/local/local.py]: (facepalm) good catch @yadudoc

**Comment by annawoodard** [comment]
Thanks for the feedback, @yadudoc! I've implemented your requests.

**Pull request merged**